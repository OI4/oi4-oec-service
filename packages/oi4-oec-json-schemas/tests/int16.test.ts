import { matchersWithOptions } from 'jest-json-schema'
import schema from '../schemas/dataTypes/int16.schema.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  })
)

describe('int16 schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect(-32768).toMatchSchema(schema)
    expect(-32767).toMatchSchema(schema)
    expect(-1).toMatchSchema(schema)
    expect(0).toMatchSchema(schema)
    expect(1).toMatchSchema(schema)
    expect(1.0).toMatchSchema(schema)
    expect(32766).toMatchSchema(schema)
    expect(32767).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    expect(-32769).not.toMatchSchema(schema)
    expect(1.5).not.toMatchSchema(schema)
    expect(32768).not.toMatchSchema(schema)
    expect(() => {
      expect(-32769).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('-32769')
    expect(() => {
      expect(1.5).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('1.5')
    expect(() => {
      expect(32768).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('32768')
  })
})
