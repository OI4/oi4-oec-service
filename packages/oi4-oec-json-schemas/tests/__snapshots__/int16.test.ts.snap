// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`int16 schema test invalid values: -32769 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must be >= -32768[39m
[31m    Path:     dataTypes/int16.schema.json#/minimum[39m
[31m[39m"
`;

exports[`int16 schema test invalid values: 1.5 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must be integer[39m
[31m    Path:     dataTypes/int16.schema.json#/type[39m
[31m[39m"
`;

exports[`int16 schema test invalid values: 32768 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must be <= 32767[39m
[31m    Path:     dataTypes/int16.schema.json#/maximum[39m
[31m[39m"
`;
